command/plugins_test.go

```diff
@@ -147,10 +148,10 @@ func (i *mockProviderInstaller) FileName(provider, version string) string {
 	return fmt.Sprintf("terraform-provider-%s_v%s_x4", provider, version)
 }
 
-func (i *mockProviderInstaller) Get(provider string, req discovery.Constraints) (discovery.PluginMeta, tfdiags.Diagnostics, error) {
+func (i *mockProviderInstaller) Get(provider addrs.ProviderConfig, req discovery.Constraints) (discovery.PluginMeta, tfdiags.Diagnostics, error) {
```

Comment:
`addrs.ProviderConfig` doesn't feel quite right to me here... that refers to a `provider` block in the configuration, of which there might be multiple with the same name. But in this case we're talking about _the_ AWS provider, regardless of any configuration blocks.

:thinking: What do you think about perhaps introducing a new struct `addrs.ProviderType` that for now might just look like this:

```go
type ProviderType struct {
  Name string
}
```

...but then later could grow to be something more like this:

```go
type ProviderType struct {
  Host      svchost.Hostname
  Namespace string
  Name      string
}
```

I'm anticipating that further down the road here we'll have some logic in core somewhere that can use per-module provider short-names to correlate each `addrs.ProviderConfig` with its associated `addrs.ProviderType`, which would then allow us to represent statements like "The provider config `module.example.realaws` is a configuration for the provider type `registry.terraform.io/hashicorp/aws`" while having the Go compiler help us keep the distinction between those straight, such that we know for any bit of code whether it's talking about the idea of the AWS provider itself or about one specific configuration of the provider.

For now, that logic would be pretty straightforward: given any `addrs.ProviderConfig`, just copy the `Type` value into the `Name` of a new `addrs.ProviderType`. But as we introduce explicit sources later, we can evolve that to do something more significant based on explicit provider source specifications in the configuration.

Reply:
Ah, ok. I was thinking that I'd add `Namespace` and `Host` to the `ProviderConfig` struct, but I see what you're saying, and it makes good sense. Thank you, I'll go update things!